src/Receiver.cc

```diff
@@ -606,32 +415,44 @@ Receiver::checkResendTimeouts(uint64_t now, MessageBucket* bucket)
         return;
     }
 
+    SpinLock::Lock lock_bucket(bucket->mutex);
```

Comment:
Curious about your thoughts behind holding the lock while processing all expired timeouts vs just one timeout at a time.

Reply:
My initial thought is that it shouldn't matter so locking once is simpler. After second thought, I think it makes more sense to lock and unlock at every iteration. I have reverted this change.